### ISSUE TITLE
lines_count should warn in case of Err

### ISSUE THREAD

**Issue by DorianCoding** [rust-lang/rust-clippy#12858]
### What it does

As stated [here](https://rust-lang.github.io/rust-clippy/master/index.html#/map_while), Lines can enter an infinite loop if nothing like `map_while` is used when used on directory. Therefore something like `lines().count` may enter an infinite loop. Therefore, it would be good if a warn can be added to fix like : `lines().map_while(Result::ok).count()`.

### Advantage

- It will avoid entering a loop when used on directory in certain platforms.

### Drawbacks

- Some users may be interested in getting the count with lines with errors included.

### Example

```rust
<code>
```

Could be written as:

```rust
let linecount = BufReader::new(File::open("some-path")?).lines().count();
```

Could be warned and this used instead :

```rust
let linecount = BufReader::new(File::open("some-path")?).lines().map_while(Result::ok).count();
```

**Comment by scottlamb** [comment]
+1 to the idea of having a lint on this. I'd prefer the suggested alternative be to bubble up the error rather than swallow it.
